apps/web/src/pages/checkout/[packSlug]/[method].tsx

```diff
@@ -0,0 +1,215 @@
+import { Environment } from '@/environment'
+import DefaultLayout from '@/layouts/default-layout'
+import {
+  getAuthenticatedUser,
+  handleUnauthenticatedRedirect,
+} from '@/services/api/auth-service'
+import CheckoutTemplate from '@/templates/checkout-method-template'
+import { urls } from '@/utils/urls'
+
+export interface CheckoutMethodPageProps {
+  address: string | null
+  auctionPackId: string | null
+  currentBid: number | null
+  release: PublishedPack
+}
+
+export default function CheckoutMethodPage({
+  address,
+  auctionPackId,
+  currentBid,
+  release,
+}: CheckoutMethodPageProps) {
+  const { t } = useTranslation()
+  const { query } = useRouter()
+  const paymentProps = usePaymentProvider({
+    auctionPackId,
+    currentBid,
+    release,
+  })
+  const { setStatus, setAddress } = paymentProps
+
+  // Set the address retrieved in server side props
+  useEffect(() => {
+    if (address) {
+      setAddress(address)
+    }
+  }, [address, setAddress])
+
+  // Set the status to the status listed as a query param:
+  useEffect(() => {
+    if (query.step) {
+      const status =
+        query?.step === 'details' ? CheckoutStatus.form : CheckoutStatus.summary
+      setStatus(status)
+    }
+  }, [query.step, setStatus])
+
+  useEffect(() => {
+    Analytics.instance.beginCheckout({
+      itemName: release.title,
+      value: currentBid ?? release.price,
+    })
+  }, [currentBid, release])
+
+  return (
+    <DefaultLayout
+      pageTitle={
+        release.type === PackType.Auction
+          ? t('common:pageTitles.Placing Bid', { name: release.title })
+          : t('common:pageTitles.Checking Out', { name: release.title })
+      }
+      panelPadding
+    >
+      <CheckoutTemplate {...paymentProps} />
+    </DefaultLayout>
+  )
+}
+
+export const getServerSideProps: GetServerSideProps<
+  CheckoutMethodPageProps
+> = async (context) => {
+  const params = context.params
+  // Verify authentication
+  const user = await getAuthenticatedUser(context)
+  if (!user) {
+    return handleUnauthenticatedRedirect(context.resolvedUrl)
+  }
+
+  // Redirect to checkout page to select method if the method isn't recognized
+  if (
+    params?.method &&
+    !Object.values(CheckoutMethod).includes(params.method as CheckoutMethod)
+  ) {
+    return {
+      redirect: {
+        destination: urls.checkoutPack.replace(
+          ':packSlug',
+          context?.params?.packSlug as string
+        ),
+        permanent: false,
+      },
+    }
+  }
+
+  // Redirect to card payment flow if feature flags aren't present
+  if (
+    !Environment.isWireEnabled &&
+    !Environment.isCryptoEnabled &&
+    params?.method !== CheckoutMethod.card
+  ) {
+    return {
+      redirect: {
+        destination: urls.checkoutPackWithMethod
+          .replace(':packSlug', context?.params?.packSlug as string)
+          .replace(':method', 'card'),
+        permanent: false,
+      },
+    }
+  }
```

Comment:
Redirects if a) the checkout method isn't recognized and b) if another method is accessed when feature flags aren't enabled.